src/api/commands/itemimage/ItemImageAddCommand.ts

```diff
@@ -76,6 +81,61 @@ export class ItemImageAddCommand extends BaseCommand implements RpcCommandInterf
         return itemImage;
     }
 
+    public async validate(data: RpcRequest): Promise<RpcRequest> {
+        // check listingItemTemplate id present in params
+        const listingItemTemplateId = data.params[0];
+        if (!listingItemTemplateId) {
+            this.log.error('ListingItemTemplate id can not be null.');
+            throw new MessageException('ListingItemTemplate id can not be null.');
+        }
+
+        // No need to validate data, it's optional and a string
+        const dataId = data.params[1];
+
+        let protocol = data.params[2];
+        if (protocol) {
+            if (!dataId) {
+                // Probably impossible, but throw an error if dataId arg (the arg required before protocol) doesn't exist.
+                throw new MessageException('Protocol arg was present, but dataId arg is missing.');
+            }
+            if (typeof protocol !== 'string') {
+                this.log.error('Protocol is present but not a string.');
+                throw new MessageException('Protocol is present but not a string.');
+            }
+            if (!ImageDataProtocolType[protocol]) {
+                this.log.error('Invalid protocol.');
+                throw new MessageException('Invalid protocol.');
+            }
+            protocol = ImageDataProtocolType[protocol];
+        }
+
+        let encoding = data.params[3];
+        if (encoding) {
+            if (!protocol) {
+                // Probably impossible, but throw an error if protocol arg (the arg required before encoding) doesn't exist.
+                throw new MessageException('Encoding arg was present, but protocol arg is missing.');
```

Comment:
its impossible, we should have allready thrown an error if protocol doesnt exists